src/picongpu/include/plugins/radiation/GetRadiationFlag.hpp

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2017 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+
+#include "traits/HasFlag.hpp"
+
+
+namespace picongpu
+{
+    /** read the `radiationFlag` of a species */
+    template< bool hasRadiationFlag >
+    struct GetRadiationFlag
+    {
+        /** get the attribute value of `radiationFlag`
+         *
+         * @param particle particle to be used
+         * @return value of the attribute `radiationFlag`
+         */
+        template< typename T_Particle >
+        HDINLINE bool operator()( const T_Particle& particle ) const
+        {
+            return particle[ radiationFlag_ ];
+        }
+    };
+
+    /** specialization
+     *
+     * specialization for the case that the species not owns the attribute
+     * `radiationFlag`
+     */
+    template< >
+    struct GetRadiationFlag< false >
+    {
+        /** get the attribute value of `radiationFlag`
+         *
+         * @param particle to be used
+         * @return always true
+         */
+        template< typename T_Particle >
+        HDINLINE bool operator()( const T_Particle& ) const
+        {
+            return true;
+        }
+    };
+
+    /** get the value of the particle attribute `radiationFlag`
+     *
+     * Allow to read out the value of the attribute `radiationFlag` also if
+     * it is not defined for the particle.
+     *
+     * @tparam T_Particle particle type
+     * @param particle valid particle
+     * @return particle attribute value `radiationFlag`, always `true` if attribute `radiationFlag` is not defined
+     */
+    template< typename T_Particle >
+    HDINLINE bool getRadiationFlag( const T_Particle& particle )
+    {
+        constexpr bool hasRadiationFlag = PMacc::traits::HasFlag<
+            typename T_Particle::FrameType,
+            radiationFlag
+        >::type::value;
+        return GetRadiationFlag< hasRadiationFlag >{}( particle );
+    }
+} // namespace picongpu
```

Comment:
is not required by our coding style^^, could I skip this? The begin of the namespace also does not contain a empty line.